nanome/_internal/_plugin_instance.py

```diff
@@ -30,6 +34,39 @@ def _call(self, id, *args):
         except KeyError:
             Logs.warning('Received an unknown callback id:', id)
 
+    @classmethod
+    def _hook_complex_updated(cls, index, complex):
+        cls.__complex_updated_callbacks[index] = complex
+
+    @classmethod
+    def _hook_selection_changed(cls, index, complex):
+        cls.__selection_changed_callbacks[index] = complex
+
+    @classmethod
+    def _on_complex_updated(cls, index, new_complex):
+        callbacks = _PluginInstance.__complex_updated_callbacks
+        try:
+            callbacks[index]._on_complex_updated(new_complex)
```

Comment:
the value in the dictionary is an object with acomplex? I don't think this is a good idea. Complexes are abandoned between frames. 

The map should just go from index to callback, that way you don't need to mess around with the garbage collection stuff either.